### DISCUSSION TITLE
Maya 2020.2 and Release 0.3.0 (windows 10) - Plugin not loading

### DISCUSSION THREAD

**Discussion by brycelynch** [Autodesk/maya-usd#766] [Q&A]
Clean install of Windows 10 Maya 2020.2 and maya-usd release version 0.3.0 fails to load plugin with following error:

**The specified procedure could not be found.**

Here is the output from console:

// Error: file: C:/Program Files/Autodesk/Maya2020/scripts/others/pluginWin.mel line 317: Unable to dynamically load : C:/Program Files/Autodesk/MayaUSD/Maya2020/0.3.0_202008140050-b9aab68/mayausd/MayaUSD/plugin/adsk/plugin/mayaUsdPlugin.mll
The specified procedure could not be found.
 // 
// Error: file: C:/Program Files/Autodesk/Maya2020/scripts/others/pluginWin.mel line 317: The specified procedure could not be found.
 (mayaUsdPlugin) // 
// Error: file: C:/Program Files/Autodesk/Maya2020/scripts/others/pluginWin.mel line 317: Unable to dynamically load : C:/Program Files/Autodesk/MayaUSD/Maya2020/0.3.0_202008140050-b9aab68/mayausd/MayaUSD/lib/maya/mtoh.mll
The specified procedure could not be found.
 // 
// Error: file: C:/Program Files/Autodesk/Maya2020/scripts/others/pluginWin.mel line 317: The specified procedure could not be found.
 (mtoh) // 

![Screenshot_11](https://user-images.githubusercontent.com/8460470/92420780-19ae5f80-f12a-11ea-8b68-d30530ca0866.jpg)

mod file located in autodesk shared common files appears to be mapped correctly to the plugin installation.

**Comment by santosg87** [comment]
HI @brycelynch do you have any environment variables from previous builds of usd? the error you have is a generic plugin load error, but it isn't uncommon for environment variables (from old usd builds, or python paths, etc) to affect this.

**Comment by santosg87** [comment]
@brycelynch we have mayaUSD 0.4.0 available now, feel free to give a try on it and see if that loads as expected for you. 

Cheers!

**Comment by DavidSpeicher** [comment]
hi im also having this issue. Maya 2020.3 i tried versions 0.3.0 , 0.4.0 and 0.5.0 all resulted with this error 
// Error: file: C:/Program Files/Autodesk/Maya2020/scripts/others/pluginWin.mel line 317: The specified procedure could not be found.
 (mayaUsdPlugin) // 
// Error: file: C:/Program Files/Autodesk/Maya2020/scripts/others/pluginWin.mel line 317: Unable to dynamically load : C:/Program Files/Autodesk/MayaUSD/Maya2020/0.4.0_202009170712-f92ca3f/mayausd/MayaUSD/plugin/adsk/plugin/mayaUsdPlugin.mll
The specified procedure could not be found.
 // 
// Error: file: C:/Program Files/Autodesk/Maya2020/scripts/others/pluginWin.mel line 317: The specified procedure could not be found.
 (mayaUsdPlugin) //

**Comment by seando-adsk** [comment]
@DavidSpeicher Make sure that you don't have any versions of USD somewhere in your PATH. For instance the "Omniverse Connect plugin for Maya" will add USD to the path which conflicts with our plugin.

**Comment by DavidSpeicher** [comment]
this was my first time installing this plugin. so i dont think that would be the issue.

**Comment by santosg87** [comment]
@DavidSpeicher from the behavior you are describing, it does sounds like you might have something conflicting with the plugin.

I would recommend running the mel command
getenv "MAYA_MODULE_PATH"

and checking within the locations that maya returns to see if you have any "USD" or mayaUSD libraries/folders. 

cheers!

**Comment by kxl-adsk** [comment]
FWIW. I converted this issue to a discussion. Once an issue is identified with repro steps, we can get it logged.

**Comment by kxl-adsk** [comment]
I would also look at the search paths and confirm that there is no other USD library in there:
```
import os
print(os.environ['PATH'])
```

**Comment by abribanez** [comment]
Hi, I also have the same issue as @DavidSpeicher and @brycelynch. Installed Maya 2020.2, then installed release 0.5.0, opened Maya, tried to load the plugins, and ran into the same errors. Checked every single location in my PATH and MAYA_MODULE_PATH, there's nothing USD-related inside those locations except for the .mods that were automatically generated by the installer (there shouldn't be anything else anyway as this is the first time I install USD)

**Reply by abribanez** [reply]
Solved it by manually adding these locations to the system PATH:
C:\Program Files\Autodesk\MayaUSD\Maya2020\0.5.0_202010161521-0a18170\mayausd\USD\bin
C:\Program Files\Autodesk\MayaUSD\Maya2020\0.5.0_202010161521-0a18170\mayausd\USD\lib

and this to the PYTHONPATH:
C:\Program Files\Autodesk\MayaUSD\Maya2020\0.5.0_202010161521-0a18170\mayausd\USD\lib\python

I tried building USD and maya-usd too. I was able to succesfully build USD but I wasn't able to build maya-usd, I kept getting a `pthread.h not found` error. Hope this helps someone. Cheers!

**Reply by kxl-adsk** [reply]
This is still surprising. What was in your `mod` file? What's in `MAYA_MODULE_PATH`?

**Reply by abribanez** [reply]
In the `MAYA_MODULE_PATH`:
`C:\Python27\Lib\site-packages`
`C:\Program Files\Autodesk\MayaUSD\Maya2020\0.5.0_202010161521-0a18170`

In the `mod` file:
```
+ USD 0.20.8-gold mayausd/USD
icons: 
plug-ins: 
presets: 
scripts: 
resources: 
PYTHONPATH+:=lib/python
PATH+:=bin
PATH+:=lib
USD_LOCATION:=

+ MayaUSD_LIB 0.5.0 mayausd/MayaUSD
icons: lib/icons
plug-ins: 
presets: 
scripts: 
resources: 
PATH+:=lib
PYTHONPATH+:=lib/python
PXR_PLUGINPATH_NAME+:=lib/usd
VP2_RENDER_DELEGATE_PROXY=1


+ MayaUSD 0.5.0 mayausd/MayaUSD/plugin/adsk
plug-ins: plugin

+ MTOH 0.5.0 mayausd/MayaUSD/lib
icons: 
plug-ins: maya
presets: 
scripts: 
resources: 
```


Also, I find out that `usdview` is not working. Maybe related to the fact that I have installed Python 2.7.15 64bits.
Here's the error that shows up:
```
Traceback (most recent call last):
  File "C:\Program Files\Autodesk\MayaUSD\Maya2020\0.5.0_202010161521-0a18170\mayausd\USD\bin\usdview", line 28, in <module>
    import pxr.Usdviewq as Usdviewq
  File "C:\Program Files\Autodesk\MayaUSD\Maya2020\0.5.0_202010161521-0a18170\mayausd\USD\lib\python\pxr\Usdviewq\__init__.py", line 29, in <module>
    from .qt import QtWidgets, QtCore
  File "C:\Program Files\Autodesk\MayaUSD\Maya2020\0.5.0_202010161521-0a18170\mayausd\USD\lib\python\pxr\Usdviewq\qt.py", line 42, in <module>
    PySideModule = GetPySideModule()
  File "C:\Program Files\Autodesk\MayaUSD\Maya2020\0.5.0_202010161521-0a18170\mayausd\USD\lib\python\pxr\Usdviewq\qt.py", line 31, in GetPySideModule
    from . import attributeValueEditorUI
  File "C:\Program Files\Autodesk\MayaUSD\Maya2020\0.5.0_202010161521-0a18170\mayausd\USD\lib\python\pxr\Usdviewq\attributeValueEditorUI.py", line 11, in <module>
    from PySide2 import QtCore, QtGui, QtWidgets
ImportError: No module named PySide2
```

Here's whats on my pip list:
```
appdirs             1.4.4
configparser        4.0.2
contextlib2         0.6.0.post1
distlib             0.3.1
filelock            3.0.12
importlib-metadata  2.0.0
importlib-resources 3.3.0
Jinja2              2.11.2
MarkupSafe          1.1.1
pathlib2            2.3.5
pip                 20.2.4
pyd                 0.13.1
PyOpenGL            3.1.5
PySide              1.2.4
scandir             1.10.0
setuptools          40.6.2
singledispatch      3.4.0.3
six                 1.15.0
typing              3.7.4.3
virtualenv          20.1.0
zipp                1.2.0
```

**Reply by seando-adsk** [reply]
I'm curious as to why you have your system python in "MAYA_MODULE_PATH"? Check your PATH and PYTHONPATH and make sure they don't have any references to your system python. You can check these from inside of Maya with the following commands. 
`getenv "PATH";`
`getenv "PYTHONPATH";`

**Reply by DavidSpeicher** [reply]
version 0.6.0 im still unable to use this plugin at all. im also able to confirm this on 4 other computers with fresh windows and only maya 2020 installed. 

// Error: file: C:/Program Files/Autodesk/Maya2020/scripts/others/pluginWin.mel line 317: Unable to dynamically load : C:/Program Files/Autodesk/MayaUSD/Maya2020/0.6.0_202012102328-efea2cd/mayausd/MayaUSD/plugin/adsk/plugin/mayaUsdPlugin.mll
The specified procedure could not be found.
 // 
// Error: file: C:/Program Files/Autodesk/Maya2020/scripts/others/pluginWin.mel line 317: The specified procedure could not be found.
 (mayaUsdPlugin) //

**Reply by DavidSpeicher** [reply]
> I'm curious as to why you have your system python in "MAYA_MODULE_PATH"? Check your PATH and PYTHONPATH and make sure they don't have any references to your system python. You can check these from inside of Maya with the following commands.
> `getenv "PATH";`
> `getenv "PYTHONPATH";`

getenv "PATH";
getenv "PYTHONPATH";
// Result: C:/ProgramData/Redshift/Plugins/Maya/Common/scripts;C:/Program Files/Autodesk/Maya2020/plug-ins/ATF/scripts;C:/Program Files/Autodesk/Bifrost/Maya2020/2.2.0.0/bifrost/scripts;C:/Program Files/Autodesk/Maya2020/plug-ins/MASH/scripts;C:/ProgramData/Autodesk/ApplicationPlugins/MayaBonusTools-2017-2020/Contents/scripts;C:/Program Files/Autodesk/MayaUSD/Maya2020/0.6.0_202012102328-efea2cd/mayausd/MayaUSD/plugin/adsk/scripts;C:/Program Files/Ephere/OrnatrixMaya2020/scripts/Autoload/AETemplates;C:/Program Files/Ephere/OrnatrixMaya2020/scripts/Autoload/Others;C:/Program Files/Ephere/OrnatrixMaya2020/scripts/Autoload/Stack;C:/Program Files/Ephere/OrnatrixMaya2020/scripts/Autoload;C:/Program Files/AMD/RadeonProRenderPlugins/Maya/scripts;C:/Program Files/Rokoko Motion Library/Maya/2020/scripts;C:/Program Files/Autodesk/Maya2020/vray/scripts;C:/Program Files/Autodesk/Maya2020/plug-ins/fbx/scripts;C:/Program Files/Autodesk/Maya2020/plug-ins/camd/scripts;C:/Program Files/Autodesk/Arnold/maya2020/scripts;C:/Program Files/Allegorithmic/Substance in Maya/2020/scripts;C:/Program Files/Autodesk/Bifrost/Maya2020/2.2.0.0/vnn/scripts;C:/Program Files/Autodesk/Maya2020/plug-ins/xgen/scripts;C:/Program Files/Autodesk/Bifrost/Maya2020/2.2.0.0/bifrost/python/site-packages;C:/Program Files/Autodesk/MayaUSD/Maya2020/0.6.0_202012102328-efea2cd/mayausd/MayaUSD/lib/python;C:/Program Files/Ephere/OrnatrixMaya2020/scripts/Python;C:/Program Files/Autodesk/MayaUSD/Maya2020/0.6.0_202012102328-efea2cd/mayausd/USD/lib/python //

**Reply by seando-adsk** [reply]
Sorry can you run just the PATH one and post the result. When you run them together it doesn't output both.

**Reply by DavidSpeicher** [reply]
> Sorry can you run just the PATH one and post the result. When you run them together it doesn't output both.
getenv "PATH";
// Result: C:/Program Files/Autodesk/Maya2020/Python/lib/site-packages/PySide2;C:/Program Files/Autodesk/Maya2020/plug-ins/ATF/ATF;C:/Program Files/Autodesk/Maya2020/plug-ins/ATF/ATF/alias;C:/Program Files/Autodesk/Maya2020/bin/Cg;C:/Program Files/Autodesk/Maya2020/bin;C:/Program Files (x86)/Razer/ChromaBroadcast/bin;C:/Program Files/Razer/ChromaBroadcast/bin;C:/Program Files (x86)/Razer Chroma SDK/bin;C:/Program Files/Razer Chroma SDK/bin;C:/Windows/system32;C:/Windows;C:/Windows/System32/Wbem;C:/Windows/System32/WindowsPowerShell/v1.0/;C:/Windows/System32/OpenSSH/;C:/Program Files (x86)/NVIDIA Corporation/PhysX/Common;C:/Windows/system32/config/systemprofile/AppData/Local/Microsoft/WindowsApps;C:/Users/pippi/AppData/Local/Microsoft/WindowsApps;C:/Program Files/NVIDIA Corporation/NVIDIA NvDLISR;C:/Program Files/Common Files/Autodesk Shared/;C:/ProgramData/Redshift/bin;C:/Program Files/Autodesk/Bifrost/Maya2020/2.2.0.0/bifrost/bin;C:/Program Files/Autodesk/Maya2020/plug-ins/MASH/bin;C:/Program Files/Autodesk/MayaUSD/Maya2020/0.7.0_202101191208-57c9fe5/mayausd/MayaUSD/lib;C:/Program Files/Ephere/OrnatrixMaya2020/bin;C:/Program Files/AMD/RadeonProRenderPlugins/Maya/bin;C:/Program Files/Autodesk/MayaUSD/Maya2020/0.7.0_202101191208-57c9fe5/mayausd/USD/bin;C:/Program Files/Autodesk/MayaUSD/Maya2020/0.7.0_202101191208-57c9fe5/mayausd/USD/lib;C:/Program Files/Autodesk/Arnold/maya2020/bin;C:/Program Files/Autodesk/Bifrost/Maya2020/2.2.0.0/vnn/bin;C:/Program Files/Autodesk/Maya2020/plug-ins/xgen/bin //

**Comment by Matt-V50** [comment]
Any solutions?
